pilot/pkg/kube/inject/app_probe.go

```diff
@@ -109,6 +109,9 @@ func convertAppProber(probe *corev1.Probe, newURL string, statusPort int) *corev
 	// Change the application container prober config.
 	c.Port = intstr.FromInt(statusPort)
 	c.Path = newURL
+	if c.Scheme == corev1.URISchemeHTTPS {
```

Comment:
I think this is worth commenting. Add some suggestion, feel free to update the wording

// For HTTPS prober, we change to HTTP, and pilot agent uses https to request application prober endpoint.
// Kubelet -> HTTP -> Pilot Agent -> HTTPS -> Application

Reply:
make sense